Bottom: 9a533c727fee11e0b3742cbc7f85c2e84624f405
Top:    b77886cf9538fafe20acbd198c08b8139191d9d8
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-05 04:47:05 +0000

Refresh of hid-gtk-gl-add-hacky-code-to-r

---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index c1d1b2f..6c95b75 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1614,9 +1614,12 @@ fill_outline_hole_cb (PLINE *pl, void *user_data)
 {
   struct outline_info *info = (struct outline_info *)user_data;
   PolygonType polygon;
+  PLINE *pl_copy = NULL;
 
-  poly_InvContour (pl);
-  polygon.Clipped = ContourToPoly (pl);
+  poly_CopyContour (&pl_copy, pl);
+  poly_InvContour (pl_copy);
+  polygon.Clipped = poly_Create ();
+  poly_InclContour (polygon.Clipped, pl_copy);
 
 //  if (polygon.Clipped->contours == NULL)
 //    return 0;
@@ -1632,12 +1635,8 @@ fill_outline_hole_cb (PLINE *pl, void *user_data)
 
   poly_FreeContours (&polygon.NoHoles);
 
-  /* Don't free the PLINE */
-  polygon.Clipped->contours = NULL;
   poly_Free (&polygon.Clipped);
 
-  poly_InvContour (pl);
-
   return 0;
 }
